server/src/main/scala/org/http4s/server/middleware/ConcurrentRequests.scala

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2013-2020 http4s.org
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.http4s.server.middleware
+
+import cats.data._
+import cats.effect._
+import cats.effect.concurrent._
+import cats.implicits._
+import org.http4s._
+import org.http4s.server._
+
+/** Middlewares for tracking the quantity of concurrent requests.
+  *
+  * These are generalized middlewares and can be used to implement metrics,
+  * logging, max concurrent requests, etc.
+  *
+  * @note The concurrent request count is decremented on the completion of the
+  *       Response body, or in the event of any error, and is guaranteed to
+  *       only occur once.
+  */
+object ConcurrentRequests {
```

Comment:
I think maybe I was expecting to see an `fs2.Signal` on this instead of those callbacks.  I don't have a feel for whether that's easier or harder to work with.